assembl/graphql/discussion.py

```diff
@@ -183,3 +183,30 @@ def mutate(root, args, context, info):
             languages=[LocalePreference(locale=x) for
                        x in discussion.discussion_locales])
         return UpdateDiscussionPreferences(preferences=discussion_pref)
+
+
+class VisitsAnalytics(graphene.ObjectType):
+
+    sum_visits_length = graphene.Int()
+    nb_pageviews = graphene.Int()
+    nb_uniq_pageviews = graphene.Int()
+
+    def query_analytics_single_metric(self, args, context, info, single_metric):
+        discussion_id = context.matchdict['discussion_id']
+        discussion = models.Discussion.get(discussion_id)
+        start = args.get('start_date') or None
+        end = args.get('end_date') or None
```

Comment:
You didn't define `start_date`, neither `end_date` as args for `sum_visits_length`, `nb_pageviews`, `nb_uniq_pageviews`.

Reply:
yes these are optional

Reply:
Currently you can't do a query 
```
query GetVisitsAnalytics {
￼  visitsAnalytics {
￼    sumVisitsLength(start_date: "2017-11-01T00:00:00Z", end_date: ""2017-12-31T29:59:59Z"")
￼    nbPageviews(start_date: "2017-11-01T00:00:00Z", end_date: ""2017-12-31T29:59:59Z"")
￼    nbUniqPageviews(start_date: "2017-11-01T00:00:00Z", end_date: ""2017-12-31T29:59:59Z"")
￼  }
}
```

If you want to do such query, you need:
```
sum_visits_length = graphene.Int(start_date=DateTime(), end_date=DateTime())
nb_pageviews = graphene.Int(start_date=DateTime(), end_date=DateTime())
nb_uniq_pageviews = graphene.Int(start_date=DateTime(), end_date=DateTime())
```

Reply:
@swergas You didn't address this one.

Reply:
Yes this was intentional. Having the right architecture for future extension and having it work from the REST API v2 is enough for now (so we can answer probable queries from clients), because setting time limits feature is out of scope of the story. But yes Maybe this is not clear enough in the code :/

Reply:
Ok. In the code, it's misleading, you thing it would work, but it will raise an error. You should put a comment in the code with a link to my previous comment for the future you. :)